pkg/rtc/subscriptionmanager_test.go

```diff
@@ -303,7 +308,9 @@ func TestSubscribeStatusChanged(t *testing.T) {
 	require.Eventually(t, func() bool {
 		return !s1.needsUnsubscribe()
 	}, subSettleTimeout, subCheckInterval, "track1 should be unsubscribed")
-	require.Equal(t, int32(1), numParticipantUnsubscribed.Load())
+	require.Eventually(t, func() bool {
+		return numParticipantUnsubscribed.Load() == 1
+	}, subSettleTimeout, subCheckInterval, "should be subscribed to publisher")
```

Comment:
This one fails once in a while because the callback happens in a goroutine. Using `Eventually` here to handle that. But, need some more work to handle things like a counter should remain 0, i. e. we can put a wait and assume that the goroutine callback did not happen and hence the test passed, but that is not great. Anyhow, for now just handling the failing paths.

Will have a look later at making the test more robust (i. e. making sure that test actually tests what it intends to test, checking for existing value without waiting for some callback goroutine to run and declaring success is not really testing that path, need to find some way to ensure that what was not supposed to happen did not really happen)